<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>TextEditor</class>
 <widget class="QDialog" name="TextEditor">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>780</width>
    <height>436</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Text Editor</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="spacing">
    <number>10</number>
   </property>
   <property name="leftMargin">
    <number>10</number>
   </property>
   <property name="topMargin">
    <number>10</number>
   </property>
   <property name="rightMargin">
    <number>10</number>
   </property>
   <property name="bottomMargin">
    <number>10</number>
   </property>
   <item>
    <widget class="QFrame" name="attr">
     <property name="minimumSize">
      <size>
       <width>760</width>
       <height>30</height>
      </size>
     </property>
     <widget class="QPushButton" name="textBgColor">
      <property name="geometry">
       <rect>
        <x>320</x>
        <y>0</y>
        <width>30</width>
        <height>30</height>
       </rect>
      </property>
      <property name="toolTip">
       <string>Text background color</string>
      </property>
      <property name="icon">
       <iconset>
        <normaloff>image/icon/color_text_background_64.png</normaloff>image/icon/color_text_background_64.png</iconset>
      </property>
     </widget>
     <widget class="QPushButton" name="right">
      <property name="geometry">
       <rect>
        <x>430</x>
        <y>0</y>
        <width>30</width>
        <height>30</height>
       </rect>
      </property>
      <property name="toolTip">
       <string>Align right</string>
      </property>
      <property name="icon">
       <iconset>
        <normaloff>image/icon/text_right_64.png</normaloff>image/icon/text_right_64.png</iconset>
      </property>
      <property name="checkable">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QFontComboBox" name="family">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>141</width>
        <height>30</height>
       </rect>
      </property>
      <property name="editable">
       <bool>false</bool>
      </property>
     </widget>
     <widget class="QPushButton" name="justify">
      <property name="geometry">
       <rect>
        <x>460</x>
        <y>0</y>
        <width>30</width>
        <height>30</height>
       </rect>
      </property>
      <property name="toolTip">
       <string>Align justify</string>
      </property>
      <property name="icon">
       <iconset>
        <normaloff>image/icon/text_justify_64.png</normaloff>image/icon/text_justify_64.png</iconset>
      </property>
      <property name="checkable">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QPushButton" name="textColor">
      <property name="geometry">
       <rect>
        <x>320</x>
        <y>0</y>
        <width>30</width>
        <height>30</height>
       </rect>
      </property>
      <property name="toolTip">
       <string>Text color</string>
      </property>
      <property name="icon">
       <iconset>
        <normaloff>image/icon/color_text_64.png</normaloff>image/icon/color_text_64.png</iconset>
      </property>
     </widget>
     <widget class="QPushButton" name="center">
      <property name="geometry">
       <rect>
        <x>400</x>
        <y>0</y>
        <width>30</width>
        <height>30</height>
       </rect>
      </property>
      <property name="toolTip">
       <string>Align center</string>
      </property>
      <property name="icon">
       <iconset>
        <normaloff>image/icon/text_center_64.png</normaloff>image/icon/text_center_64.png</iconset>
      </property>
      <property name="checkable">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QPushButton" name="italic">
      <property name="geometry">
       <rect>
        <x>230</x>
        <y>0</y>
        <width>30</width>
        <height>30</height>
       </rect>
      </property>
      <property name="toolTip">
       <string>Italic</string>
      </property>
      <property name="icon">
       <iconset>
        <normaloff>image/icon/text_italic_64.png</normaloff>image/icon/text_italic_64.png</iconset>
      </property>
      <property name="checkable">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QPushButton" name="bold">
      <property name="geometry">
       <rect>
        <x>200</x>
        <y>0</y>
        <width>30</width>
        <height>30</height>
       </rect>
      </property>
      <property name="toolTip">
       <string>Bold</string>
      </property>
      <property name="icon">
       <iconset>
        <normaloff>image/icon/text_bold_64.png</normaloff>image/icon/text_bold_64.png</iconset>
      </property>
      <property name="checkable">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QPushButton" name="left">
      <property name="geometry">
       <rect>
        <x>370</x>
        <y>0</y>
        <width>30</width>
        <height>30</height>
       </rect>
      </property>
      <property name="toolTip">
       <string>Align left</string>
      </property>
      <property name="icon">
       <iconset>
        <normaloff>image/icon/text_left_64.png</normaloff>image/icon/text_left_64.png</iconset>
      </property>
      <property name="checkable">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QPushButton" name="underLine">
      <property name="geometry">
       <rect>
        <x>260</x>
        <y>0</y>
        <width>30</width>
        <height>30</height>
       </rect>
      </property>
      <property name="toolTip">
       <string>Underline</string>
      </property>
      <property name="icon">
       <iconset>
        <normaloff>image/icon/text_underline_64.png</normaloff>image/icon/text_underline_64.png</iconset>
      </property>
      <property name="checkable">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QDoubleSpinBox" name="letterSpace">
      <property name="geometry">
       <rect>
        <x>530</x>
        <y>0</y>
        <width>50</width>
        <height>30</height>
       </rect>
      </property>
      <property name="toolTip">
       <string>Letter space</string>
      </property>
      <property name="decimals">
       <number>0</number>
      </property>
      <property name="maximum">
       <double>512.000000000000000</double>
      </property>
     </widget>
     <widget class="QDoubleSpinBox" name="lineSpace">
      <property name="geometry">
       <rect>
        <x>620</x>
        <y>0</y>
        <width>50</width>
        <height>30</height>
       </rect>
      </property>
      <property name="toolTip">
       <string>Line space</string>
      </property>
      <property name="decimals">
       <number>0</number>
      </property>
      <property name="maximum">
       <double>512.000000000000000</double>
      </property>
     </widget>
     <widget class="QDoubleSpinBox" name="paragraphSpace">
      <property name="geometry">
       <rect>
        <x>710</x>
        <y>0</y>
        <width>50</width>
        <height>30</height>
       </rect>
      </property>
      <property name="toolTip">
       <string>Paragraph space</string>
      </property>
      <property name="decimals">
       <number>0</number>
      </property>
      <property name="maximum">
       <double>512.000000000000000</double>
      </property>
     </widget>
     <widget class="QPushButton" name="letterSpaceIcon">
      <property name="geometry">
       <rect>
        <x>500</x>
        <y>0</y>
        <width>30</width>
        <height>30</height>
       </rect>
      </property>
      <property name="icon">
       <iconset>
        <normaloff>image/icon/text_letter_space_64.png</normaloff>image/icon/text_letter_space_64.png</iconset>
      </property>
      <property name="flat">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QPushButton" name="lineSpaceIcon">
      <property name="geometry">
       <rect>
        <x>590</x>
        <y>0</y>
        <width>30</width>
        <height>30</height>
       </rect>
      </property>
      <property name="icon">
       <iconset>
        <normaloff>image/icon/text_line_space_64.png</normaloff>image/icon/text_line_space_64.png</iconset>
      </property>
      <property name="flat">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QPushButton" name="paragraphSpaceIcon">
      <property name="geometry">
       <rect>
        <x>680</x>
        <y>0</y>
        <width>30</width>
        <height>30</height>
       </rect>
      </property>
      <property name="icon">
       <iconset>
        <normaloff>image/icon/text_paragraph_space_64.png</normaloff>image/icon/text_paragraph_space_64.png</iconset>
      </property>
      <property name="flat">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QSpinBox" name="point">
      <property name="geometry">
       <rect>
        <x>140</x>
        <y>0</y>
        <width>51</width>
        <height>30</height>
       </rect>
      </property>
      <property name="toolTip">
       <string>Font size</string>
      </property>
      <property name="maximum">
       <number>512</number>
      </property>
      <property name="value">
       <number>10</number>
      </property>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QScrollArea" name="textScroll">
     <property name="palette">
      <palette>
       <active>
        <colorrole role="Base">
         <brush brushstyle="SolidPattern">
          <color alpha="255">
           <red>51</red>
           <green>51</green>
           <blue>51</blue>
          </color>
         </brush>
        </colorrole>
        <colorrole role="Window">
         <brush brushstyle="SolidPattern">
          <color alpha="255">
           <red>0</red>
           <green>0</green>
           <blue>0</blue>
          </color>
         </brush>
        </colorrole>
       </active>
       <inactive>
        <colorrole role="Base">
         <brush brushstyle="SolidPattern">
          <color alpha="255">
           <red>51</red>
           <green>51</green>
           <blue>51</blue>
          </color>
         </brush>
        </colorrole>
        <colorrole role="Window">
         <brush brushstyle="SolidPattern">
          <color alpha="255">
           <red>0</red>
           <green>0</green>
           <blue>0</blue>
          </color>
         </brush>
        </colorrole>
       </inactive>
       <disabled>
        <colorrole role="Base">
         <brush brushstyle="SolidPattern">
          <color alpha="255">
           <red>0</red>
           <green>0</green>
           <blue>0</blue>
          </color>
         </brush>
        </colorrole>
        <colorrole role="Window">
         <brush brushstyle="SolidPattern">
          <color alpha="255">
           <red>0</red>
           <green>0</green>
           <blue>0</blue>
          </color>
         </brush>
        </colorrole>
       </disabled>
      </palette>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Plain</enum>
     </property>
     <property name="alignment">
      <set>Qt::AlignHCenter|Qt::AlignTop</set>
     </property>
     <widget class="RichEditor" name="text">
      <property name="geometry">
       <rect>
        <x>251</x>
        <y>0</y>
        <width>256</width>
        <height>192</height>
       </rect>
      </property>
     </widget>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <property name="spacing">
      <number>10</number>
     </property>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="ok">
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>30</height>
        </size>
       </property>
       <property name="text">
        <string>OK</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="cancel">
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>30</height>
        </size>
       </property>
       <property name="text">
        <string>Cancel</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>RichEditor</class>
   <extends>QTextEdit</extends>
   <header>RichEditor.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
